tests/services/upload/test_coverage_file_finder.py

```diff
@@ -153,179 +154,226 @@ def test_find_coverage_files_test_results(self, tmp_path):
             UploadCollectionResultFile(
-                Path(f"{self.project_root}/subdirectory/test_coverage.xml")
+                Path(f"{project_root}/subdirectory/test_coverage.xml")
             ),
-            UploadCollectionResultFile(Path(f"{self.project_root}/test_file.abc")),
+            UploadCollectionResultFile(Path(f"{project_root}/test_file.abc")),
             UploadCollectionResultFile(
-                Path(f"{self.project_root}/subdirectory/another_file.abc")
+                Path(f"{project_root}/subdirectory/another_file.abc")
             ),
         ]
         expected_paths = sorted([file.get_filename() for file in expected])
-        self.assertEqual(result, expected_paths)
+        assert result == expected_paths
 
-    def test_find_coverage_files_with_user_specified_files_not_found(self):
-        # Create some sample coverage files
+    def test_find_coverage_files_with_user_specified_files_not_found(
+        self, coverage_file_finder_fixture
+    ):
+        (
+            project_root,
+            coverage_file_finder,
+        ) = coverage_file_finder_fixture
+
+        # Create some sample coverage coverage_file_finder_fixture
         coverage_files = [
-            self.project_root / "coverage.xml",
-            self.project_root / "subdirectory" / "test_coverage.xml",
-            self.project_root / ".tox" / "another_file.abc",
+            project_root / "coverage.xml",
+            project_root / "subdirectory" / "test_coverage.xml",
+            project_root / ".tox" / "another_file.abc",
         ]
-        (self.project_root / "subdirectory").mkdir()
-        (self.project_root / ".tox").mkdir()
+        (project_root / "subdirectory").mkdir()
+        (project_root / ".tox").mkdir()
         for file in coverage_files:
             file.touch()
 
         # Add a non-existent file to explicitly_listed_files
-        self.coverage_file_finder.explicitly_listed_files.append(
-            self.project_root / "non_existent.xml"
+        coverage_file_finder.explicitly_listed_files.append(
+            project_root / "non_existent.xml"
         )
 
         result = sorted(
-            [file.get_filename() for file in self.coverage_file_finder.find_files()]
+            [file.get_filename() for file in coverage_file_finder.find_files()]
         )
 
         expected = [
-            UploadCollectionResultFile(Path(f"{self.project_root}/coverage.xml")),
+            UploadCollectionResultFile(Path(f"{project_root}/coverage.xml")),
             UploadCollectionResultFile(
-                Path(f"{self.project_root}/subdirectory/test_coverage.xml")
+                Path(f"{project_root}/subdirectory/test_coverage.xml")
             ),
         ]
         expected_paths = sorted([file.get_filename() for file in expected])
-        self.assertEqual(result, expected_paths)
+        assert result == expected_paths
 
-    def test_find_coverage_files_with_user_specified_files_in_default_ignored_folder(
```

Comment:
@joseph-sentry please don't remove this test, this was broken for one of our users